### PR TITLE
Adds Ini_set documentation

### PR DESCRIPTION
**Pull request by NielsdeBlaauw**
Related to #1722

### CODE REVIEW

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<documentation title="Detect incorrect use of the `ini_set()` function.">
+    <standard>
+    <![CDATA[
+      Using ini_set() for alternating PHP settings, as well as other functions with ability to change configuration at runtime of your scripts is discouraged. Changing runtime configuration might break other plugins and themes.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="ini_set() for a possibly breaking setting">
+        <![CDATA[
+ini_set('short_open_tag', 'off');
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Please add typical WP spacing around parameters.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    </code_comparison>
+    <standard>
+    <![CDATA[
+      For some configuration values there are WordPress functions. These are preferred.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="ini_set() to alter memory limits.">
+        <![CDATA[
+ini_set('memory_limit', '256M');
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Please add typical WP spacing around parameters.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<documentation title="Detect incorrect use of the `ini_set()` function.">
+    <standard>
+    <![CDATA[
+      Using ini_set() for alternating PHP settings, as well as other functions with ability to change configuration at runtime of your scripts is discouraged. Changing runtime configuration might break other plugins and themes.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="ini_set() for a possibly breaking setting">
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Please start with `Valid: ` and end with a full-stop.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        <![CDATA[
+ini_set( 'short_open_tag', 'off' );
+        ]]>
+        </code>
+        <code title="ini_set() for a possibly breaking setting">
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Please start with `Invalid: ` and end with a full-stop.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        <![CDATA[
+// ini_set should not be used.
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Add the `()` after `ini_set`.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+      For some configuration values there are WordPress functions. These are preferred.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="ini_set() to alter memory limits.">
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: I think this _invalid_ `code` block should be after the other one.

Please also start with `Invalid: `.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        <![CDATA[
+ini_set( 'memory_limit', '256M' );
+        ]]>
+        </code>
+        <code title="WordPress functional alternative">
```
Comment (GaryJones) [inline WordPress/Docs/PHP/IniSetStandard.xml]: Please start with `Valid: ` and end with a full-stop.

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 8]
```diff
@@ -0,0 +1,36 @@
+<documentation title="Detect incorrect use of the `ini_set()` function.">
+    <standard>
+    <![CDATA[
+      Using ini_set() for alternating PHP settings, as well as other functions with ability to change configuration at runtime of your scripts is discouraged. Changing runtime configuration might break other plugins and themes.
```
Comment (jrfnl) [inline WordPress/Docs/PHP/IniSetStandard.xml]: The grammar of this phrase seemed off. I've made it a bit shorter now as well.
```suggestion
      Using ini_set() and similar functions for altering PHP settings at runtime is discouraged. Changing runtime configuration might break other plugins and themes, and even WordPress itself.
```

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+<documentation title="Detect incorrect use of the `ini_set()` function.">
```
Comment (jrfnl) [inline WordPress/Docs/PHP/IniSetStandard.xml]: * No `.` at the end of the title
* Keeping it short. This is a title, not a description.
```suggestion
<documentation title="Detect use of `ini_set()`">
```

File: WordPress/Docs/PHP/IniSetStandard.xml [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        </code>
+        <code title="Invalid: ini_set() for a possibly breaking setting.">
+        <![CDATA[
+ini_set( <em>'short_open_tag'</em>, 'off' );
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+      For some configuration values there are WordPress functions. These are preferred.
```
Comment (jrfnl) [inline WordPress/Docs/PHP/IniSetStandard.xml]: There is only one config setting for which we suggest a WP function alternative.
For all the others, we suggest either a PHP function or using a WP constant.

I've tried to adjust the sentence to better reflect that.

```suggestion
      For some configuration values there are alternative ways available - either via WordPress native functionality of via standard PHP - to achieve the same without the risk of breaking interoperability. These alternatives are preferred.
```

**Comment by NielsdeBlaauw** [comment]
Completely agree with your suggestions, as these make the desired behaviour much clearer 👍

**Pull request merged**